libs/deal-ticket/src/use-order-submit.ts

```diff
@@ -35,12 +36,14 @@ const ORDER_EVENT_SUB = gql`
   }
 `;
 
-interface UseOrderSubmitMarket {
-  id: string;
-  decimalPlaces: number;
-}
+const calculateOrderSize = (
+  market: DealTicketQuery_market,
```

Comment:
chore (blocking): take a number rather than the whole markt object, as GQL we request different subsets of the market data this will likely lead to type collisions later

Reply:
looking at this again it seems quite redundant as it's just a proxy function for the `removeDecimals`. removed this for now.
question tho: not entirely sure how the type generators work, but I'd expect them to generate the correct types based on the queries we write. the market query's generated type is `DealTicketQuery_market`, shouldn't this already cover all the variations we're requesting? i was assuming it does, hence the removal of the `UseOrderSubmitMarket` type.

Reply:
So the types will generate based on the query that is specified so for example if query A chooses feild X, Y a type will be generate to reflect that. The core tenant of GQL is to only request the data you need not everything unlike rest so query B may require fields Y and Z. So we end up with 2 types depending on what was requested, and it is very unlikely that the these types will overlap.

So when we generate functions that will act on say a market the best thing to do is create an interface that requires the specific fields that the function needs e.g.

```
getMarketLeverageName (market: ILeverageMarket) { return `${market.name} x${1/market.riskFactor}` }
```
with interface
```
ILeverageMarket {
riskFactor: string
name: string
}
```

or generate just take number parameters if feesable (as in this example).

Now if we generate a different query for the same entity that creates a different type we can still use this function even if the types do not overlap perfectly, as long as the 2 fields that the function requires takes. I anticipate we will request many different versions of markets, so we want this to be reusable.

More over this makes the boundary very defined between libraries and defines a specfic interface rather than a leeky abrstraction based on what is requested which is also good IMO.

Reply:
Hopefully that all makes sense :joy:

Reply:
it does, more or less. i think what makes it a little confusing is that currently i see only one source query for getting markets in this lib, so they seemingly always use the same type among the components. anyhow, this needs another look i think, did a pretty big refactor 😄